pkg/checkpoint/checkpoint.go

```diff
@@ -3,7 +3,7 @@ package checkpoint
 import (
 	"fmt"
 
-	"google.golang.org/protobuf/proto"
+	"github.com/fxamacker/cbor/v2"
```

Comment:
I know that is still "half-baked" and it requires some type conversions to make it work, but why not using `github.com/whyrusleeping/cbor` which is the one used by all the Filecoin ecosystem?

Reply:
Yes I saw you are using the other one, but I found it quite hard to use compared to the `fxamacker` one.

- I couldn't find proper documentation for the `whyrusleeping` one
- The interface seems to be very low-level. At least your code seems to explicitly deal with types of the serialized fields. The `fxamacker` version just directly serializes the struct.
- `whyrysleeping` is at least 5 years old (which might be a good sign for stability), but `fxamacker` is a much more recent.
- `fxamacker` has decent documentation and a very simple interface.

I'm actually wondering why the rest of the Filecoin ecosystem doesn't use the newer one :-D (But I'm ready to be educated :) )